packages/experiments/src/components/CollapsibleSection/CollapsibleSection.state.tsx

```diff
@@ -1,21 +1,23 @@
 import * as React from 'react';
-import { ICollapsibleSectionProps, ICollapsibleSectionStyles } from './CollapsibleSection.types';
+import { ICollapsibleSectionProps, ICollapsibleSectionViewProps } from './CollapsibleSection.types';
 import { createRef } from 'office-ui-fabric-react';
-import { IStateProps } from '../../Foundation';
-import { getRTL, KeyCodes } from '../../Utilities';
+import { IStateComponentProps } from '../../Foundation';
+import { BaseComponent, getRTL, KeyCodes } from '../../Utilities';
 
 export interface ICollapsibleSectionState {
   collapsed: boolean;
 }
 
-// TODO: these types seem duplicated from createComponent. try to avoid duplication and be aware of circular references with IStateComponent
```

Comment:
>// TODO: [](start = 0, length = 8)

Will all of the TODOs in these files be resolved in future PRs?

Reply:
I consider these TODOs important enough to resolve before promoting this PR to production use, not necessarily within this PR. I tagged them out in the hopes of getting team feedback and consensus. I've made far more decisions on design and API than I've tagged, but I'm still looking for calibration in these core areas

Reply:
This TODO was resolved btw 😀

Reply:
I had an idea overnight, I'll write up an RFC issue highlighting goals and design. I think that'll help people review this PR quite a bit

Reply:
Created #5721